### ISSUE TITLE
Add option to undo "remove failed uploads" if selected accidentally

### ISSUE THREAD

**Issue by e-alfred** [nextcloud/android#1568]
### Actual behaviour
- If an upload fails for whatever reason and "remove failed uploads" is tapped accidentally instead of "restart failed uploads", there is no easy way to revert this operation. Instead all failed uploads have to be found (no list or anything, which means it is very difficult if it happens for a lot of files) and selected by hand for uploading.

### Expected behaviour
- An undo button (maybe at the bottom) should be shown like Nextcloud does everywhere else because it is good UI/UX design.
 
### Steps to reproduce
1. Generate an upload that fails (enabling instant upload and taking a picture with the camera while losing connection for example)
2. Retry to upload the image but instead accidentally tap the "remove failed uploads" in the menu instead of "retry failed uploads"

**Comment by AndyScherzinger** [comment]
Hi @e-alfred did you try the latest release candidates of 2.0.0? Just asking since this is basically fixed implicitly due to the fact that 2.0.0 doesn't have these menu items anymore but does auto retries itself 😃

**Comment by e-alfred** [comment]
The Nextcloud app version is 2.0.0 RC6, but the menu items are still there. I had to upgrade because of this bug (unrelated to the problem here): https://help.nextcloud.com/t/webdav-file-upload-fails-exception-sabre-dav-exception-notfound/5824

To test the auto retry I simply disabled mobile data and it tried to reupload a few times in short intervals, but after reenabling mobile data after some time (around 15 minutes) it doesn't do an automatic retry anymore. "Restart failed uploads" is exactly above "Remove failed uploads" in the ... menu. If I tap "restart failed uploads" it uploads them if the connection works.

I think there are two different problems as far as I can see:

- Auto retry does not work reliable enough. Maybe it should be toggled again if a working internet connection is detected (like mobile network is connected again after leaving an uncovered area or the user enables mobile data/WIFI again)?

- "Remove failed uploads" should have an undo option so the user doesn't have to fiddle around with searching all his files he wanted to auto upload. For example, making a lot of pictures and having to search/select all of the missing ones because of a single tap on the wrong button is a very inconvenient hassle.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @mario any ideas? I am running 2.0.0RC6+ (actual master) on my phone and I do not have a menu thus @e-alfred shouldn't see one either?!

**Comment by e-alfred** [comment]
Here is a redacted screenshot of the menu I can see:

![screenshot_20170921-133251 edit](https://user-images.githubusercontent.com/10073629/30708004-9141435a-9efd-11e7-9cb6-b15bf5913293.png)

Phone specs: BQ X5 Plus with Android 7.1.1

**Comment by AndyScherzinger** [comment]
Ah, right. You activated the expert mode in the settings 😃

**Comment by AndyScherzinger** [comment]
Well, anyways. Afaik we @mario, @tobiasKaminsky and I discussed to drop the complete menu after 2.0.0 as in 2.1 or 2.2.
Can you open another issue since as you mentioned the failed uploads won't restart, while you mentioned you re-enabled mobile data,, are the auto upload settings set to "wifi only" and are you then connected to a wifi? B/c this usually worked for me qzuite reliable :/

**Comment by mario** [comment]
@AndyScherzinger granted, the receiver only fires if you go to wifi, NOT when you connect in general. Maybe that's the issue?

@e-alfred are you saying you have no internet connectivity at all, and when you turn 3g/4g it doesn't start the uploads?

**Comment by e-alfred** [comment]
@AndyScherzinger @mario I disabled "Upload using WIFI only" in the settings. I was not connected to WIFI when I faced/tested this. It "auto retry" should be triggered if this setting is disabled AND the user reenables the mobile data connection or comes back into an area where mobile data is available again.

@AndyScherzinger Exactly. I took a picture while not connected at all and later on I reenabled mobile data again.
